### PR TITLE
feat(dockerhub): publish rootless containers

### PR DESCRIPTION
**Pull request by stefreak**
**What this PR does / why we need it**:
Add rootless container variant for use in environments where
using the root user is not possible, for example GitLab's Runner Operator

Supersedes #3865

**Which issue(s) this PR fixes**:

Addresses #3993

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by stefreak** [comment]
TODO: Sync with @worldofgeese to find out if these rootless images work and help make things easier in a rootless environment.

File: docs/reference/dockerhub-containers.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+| [`gardendev/garden-aws-gcloud-azure`](https://hub.docker.com/r/gardendev/garden-aws-gcloud-azure) | Contains the Garden CLI, the Google Cloud CLI and the AWS CLI v2 |
+| [`gardendev/garden-full`](https://hub.docker.com/r/gardendev/garden-full) | DEPRECATED: This container image is not being maintained anymore and will be removed in the future. |
+
+### Tags
+
+| Tag name                          | Meaning                                                          |
+|-----------------------------------|------------------------------------------------------------------|
+| `latest`                          | Latest stable release of Garden CLI 0.13 (Codename Bonsai), the container is based on Alpine Linux.
+| `bonsai-*` or `0.13*`             | Garden CLI version is 0.13 (Codename Bonsai). If the tag name does not contain edge, this is the latest stable release. |
+| `acorn-*` or `0.12*`              | Garden CLI version is 0.13 (Codename Bonsai). If the tag name does not contain edge, this is the latest stable release. |
```
Comment (stefreak) [inline docs/reference/dockerhub-containers.md:30]:
```suggestion
| `acorn-*` or `0.12-*`              | Garden CLI version is 0.12 (Codename Acorn). If the tag name does not contain edge, this is the latest stable release. |
```

File: docs/reference/dockerhub-containers.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+---
+order: 60
+title: Garden Containers on DockerHub
+---
+
+# Garden containers on DockerHub
+
+We publish a number of containers to our official DockerHub account ([hub.docker.com/gardendev](https://hub.docker.com/gardendev)).
```
Comment (stefreak) [inline docs/reference/dockerhub-containers.md:8]: Broken link
```suggestion
We publish a number of containers to our official DockerHub account ([hub.docker.com/u/gardendev](https://hub.docker.com/u/gardendev)).
```

File: support/alpine.Dockerfile [hunk 3]
```diff
@@ -31,10 +36,22 @@ RUN chmod +x /garden/garden \
   && ln -s /garden/garden /usr/local/bin/garden \
   && chmod +x /usr/local/bin/garden \
   && cd /garden/static \
-  && GARDEN_DISABLE_ANALYTICS=true GARDEN_DISABLE_VERSION_CHECK=true garden util fetch-tools --all --garden-image-build
+  && git init
```
Comment (stefreak) [inline support/alpine.Dockerfile:39]: Ownership of the git repo in garden static is not quite right yet. 
Getting this error when using the rootless container:

```
✖ Unexpected Git error occurred while running 'git status' from path "/garden/static". Exit code: 128. Error message: fatal: detected dubious ownership in repository at '/garden/static'
To add an exception for this directory, call:

	git config --global --add safe.directory /garden/static
✖ Command "git status" failed with code 128:

   fatal: detected dubious ownership in repository at '/garden/static'
To add an exception for this directory, call:

	git config --global --add safe.directory /garden/static
```

File: docs/README.md [hunk 4]
```diff
@@ -168,6 +168,7 @@
 * [Module Template Configuration](./reference/module-template-config.md)
 * [RenderTemplate Reference](./reference/render-template-config.md)
 * [Workflow Configuration](./reference/workflow-config.md)
+* [Garden Containers on DockerHub](./reference/dockerhub-containers.md)
```
Comment (worldofgeese) [inline docs/README.md:171]:
```suggestion
* [Garden Containers on Docker Hub](./reference/dockerhub-containers.md)
```
Docker Hub naming convention is to put a space between "Docker" and "Hub"

https://www.docker.com/products/docker-hub/

File: docs/reference/dockerhub-containers.md (hunk 2 above)
Comment (worldofgeese) [inline docs/reference/dockerhub-containers.md]:
```suggestion
We publish a number of containers to our official Docker Hub account ([hub.docker.com/gardendev](https://hub.docker.com/u/gardendev)).
```

Space between "Docker" and "Hub"

File: docs/reference/dockerhub-containers.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+| Container                         | Contents                                                         |
+|-----------------------------------|------------------------------------------------------------------|
+| [`gardendev/garden`](https://hub.docker.com/r/gardendev/garden) | Contains only Garden CLI                                         |
+| [`gardendev/garden-aws`](https://hub.docker.com/r/gardendev/garden-aws) | Contains the Garden CLI, and the AWS CLI v2                      |
+| [`gardendev/garden-azure`](https://hub.docker.com/r/gardendev/garden-azure) | Contains the Garden CLI, and the Azure CLI                       |
+| [`gardendev/garden-gcloud`](https://hub.docker.com/r/gardendev/garden-gcloud) | Contains the Garden CLI, and the Google Cloud CLI                |
+| [`gardendev/garden-aws-gcloud`](https://hub.docker.com/r/gardendev/garden-aws-gcloud) | Contains the Garden CLI, the Google Cloud CLI and the AWS CLI v2 |
+| [`gardendev/garden-aws-gcloud-azure`](https://hub.docker.com/r/gardendev/garden-aws-gcloud-azure) | Contains the Garden CLI, the Google Cloud CLI and the AWS CLI v2 |
```
Comment (worldofgeese) [inline docs/reference/dockerhub-containers.md:21]:
```suggestion
| [`gardendev/garden-aws-gcloud-azure`](https://hub.docker.com/r/gardendev/garden-aws-gcloud-azure) | Contains the Garden CLI, the Google Cloud CLI, the AWS CLI v2, and the Azure CLI |
```

File: support/alpine.Dockerfile (hunk 3 above)
Comment (worldofgeese) [reply support/alpine.Dockerfile:39]: Why does Garden init a git repository at `static`?

**Review by worldofgeese — CHANGES_REQUESTED** [review]

**Review by TimBeyer — APPROVED** [review]
👏

**Pull request merged**